### ISSUE TITLE
Show non-resolvable domains with a different color in the UI

### ISSUE THREAD

**Issue by ccoager** [uBlockOrigin/uBlock-issues#2484]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

This is an enhancement request - Can we modify the UI to show non-resolvable domains with a different color?  For those that block domains with Pi-hole and also use uBlock Origin medium/hard mode would find this beneficial.  When examining the domain list in the overview panel, any domain that doesn't resolve would show up in a different color so you wouldn't bother trying to unblock these domains when trying to fix a broken webpage.

### A specific URL where the issue occurs.

https://www.msn.com/

### Steps to Reproduce

1. Click on the uBlock Origin toolbar
2. Examine the domain list

### Expected behavior

CNAME's are colored blue which is handy, but it's not obvious which domains resolve and which don't.  Can we colorize the non-resolvable domains?

### Actual behavior

The domain list doesn't distinguish domains that don't resolve.

### uBO version

1.46.0

### Browser name and version

Firefox 109.0.1

### Operating System and version

Windows 10

**Comment by gothic-bum** [comment]
uBO does not perform domain name resolution. To determine if a domain is valid, it would need to be unblocked and a network request made to it.

**Comment by gwarser** [comment]
>any domain that doesn't resolve would show up in a different color so you wouldn't bother trying to unblock these domains when trying to fix a broken webpage.

You will not know it resolves until you unblock. There is no connection to remote server when something is bloked.

**Comment by ccoager** [comment]
According to the wiki:
```
"Access IP address and hostname information"

Related permission: dns.

Since version 1.25.0 (Firefox 60+ only).

This warning is triggered by the dns permission, which allows using the browser.dns API. The purpose is for uBO to gain the ability to reveal the canonical name of aliased hostnames.

Note that even without this permission, uBO can see the IP address and hostname information, through the browser.webRequest API which uBO already requires.
```
If uBO is using DNS to attempt to discover canonical names, it would also discover non-resolvable domains without needing to unblock.

**Comment by gwarser** [comment]
CNAMEs are resolved for unblocked requests. 

It might be useful to mark failed unblocked requests.

**Comment by ccoager** [comment]
> CNAMEs are resolved for unblocked requests.
> 
> It might be useful to mark failed unblocked requests.

I don't believe that is true, see example below.  CNAME "cdn.cnn.com" is resolved to "e12596.dscj.akamaiedge.net" and you can see the request is blocked because it is in red.
![cname resolution - cnn com](https://user-images.githubusercontent.com/2500709/218124046-506cfd4d-27ba-45c1-9fa6-365d174e31d4.png)

> @ccoager dns permission is only available for Firefox.

I'm aware and that is what I'm testing on.

**Comment by gorhill** [comment]
> I don't believe that is true

There were network requests for `cdn.cnn.com`, hence a DNS resolution occurred. It won't occur if `cdn.cnn.com` is wholly blocked.

> dns permission is only available for Firefox.

It does not matter, the feature request here is not really accurate, I consider that it should simply be to offer a visual hint for when a network request is blocked outside uBO (be it through pi-hole or another extension), and uBO can detect this by listening to [webRequest.onErrorOccurred](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onErrorOccurred), this does not require to deal with DNS resolution.

**Comment by garry-ut99** [comment]
Maybe better don't touch vertical lines as they indicate whether the connection passed trought uBO, and instead, just to change font's color of domain's name, just like in case of cname, proposed: `red` as the opposite to `blue` of cname.
